UWP/UWPHelpers/StorageManager.cpp

```diff
@@ -0,0 +1,331 @@
+	ExecuteTask(tmpFile, ApplicationData::Current->TemporaryFolder->CreateFileAsync(ToPlatformString(name), CreationCollisionOption::GenerateUniqueName));
+	if (tmpFile != nullptr) {
+		return FromPlatformString(tmpFile->Path);
+	}
+	else {
+		return "";
+	}
+}
+std::string GetPicturesFolder() {
+	// Requires 'picturesLibrary' capability
+	return FromPlatformString(KnownFolders::PicturesLibrary->Path);
+}
+std::string GetVideosFolder() {
+	// Requires 'videosLibrary' capability
+	return FromPlatformString(KnownFolders::VideosLibrary->Path);
+}
+std::string GetDocumentsFolder() {
+	// Requires 'documentsLibrary' capability
+	return FromPlatformString(KnownFolders::DocumentsLibrary->Path);
+}
+std::string GetMusicFolder() {
+	// Requires 'musicLibrary' capability
+	return FromPlatformString(KnownFolders::MusicLibrary->Path);
+}
+std::string GetPreviewPath(std::string path) {
+	std::string pathView = path;
+
+	pathView = ReplaceAll(pathView, "/", "\\");
+	pathView = ReplaceAll(pathView, GetLocalFolder(), "LocalState");
+	pathView = ReplaceAll(pathView, GetTempFolder(), "TempState");
+	pathView = ReplaceAll(pathView, GetInstallationFolder(), "Installation folder");
+	return pathView;
+}
+bool isLocalState(std::string path) {
+	return !_stricmp(GetPreviewPath(path).c_str(), "LocalState");
+}
+#pragma endregion
+
+#pragma region Internal
+Path PathResolver(Path path) {
+	auto root = path.GetDirectory();
+	auto newPath = path.ToString();
+	if (path.IsRoot() || !_stricmp(root.c_str(), "/") || !_stricmp(root.c_str(), "\\")) {
+		// System requesting file from app data
+		newPath = ReplaceAll(newPath, "/", (GetLocalFolder() + (path.size() > 1 ? "/" : "")));
+	}
+	return Path(newPath);
+}
+Path PathResolver(std::string path) {
+	return PathResolver(Path(path));
+}
+
+std::string ResolvePathUWP(std::string path) {
+	return PathResolver(path).ToString();
+}
+#pragma endregion
+
+#pragma region Functions
+std::map<std::string, bool> accessState;
+bool CheckDriveAccess(std::string driveName) {
+	bool state = false;
+
+	HANDLE searchResults;
+	WIN32_FIND_DATA findDataResult;
+	auto keyIter = accessState.find(driveName);
+	if (keyIter != accessState.end()) {
+		state = keyIter->second;
+	}
+	else {
+		try {
+			wchar_t* filteredPath = _wcsdup(ConvertUTF8ToWString(driveName).c_str());
+			wcscat_s(filteredPath, sizeof(L"\\*.*"), L"\\*.*");
+
+			searchResults = FindFirstFileExFromAppW(
+				filteredPath, FindExInfoBasic, &findDataResult,
+				FindExSearchNameMatch, NULL, 0);
+
+			state = searchResults != NULL && searchResults != INVALID_HANDLE_VALUE;
+			if (state) {
+				FindClose(searchResults);
+			}
+			// Cache the state
+			accessState.insert(std::make_pair(driveName, state));
+		}
+		catch (...) {
+		}
+	}
+	if (!state) {
+		state = IsRootForAccessibleItems(driveName);
+	}
+	return state;
+}
+
+bool isWriteMode(const char* mode) {
+	return (!strcmp(mode, "w") || !strcmp(mode, "wb") || !strcmp(mode, "wt") || !strcmp(mode, "at") || !strcmp(mode, "a"));
```

Comment:
These are more or less "flags", so the order is not actually fixed.  I recommend instead using:

```c++
return strpbrk(mode, "wa+") != nullptr;
```

Which is roughly equivalent to the more obviously named:
```c++
return std::string(mode).first_first_of("wa") != std::string::npos;
```